app/javascript/src/ChangePassword/components/ChangePasswordHooks.jsx

```diff
@@ -0,0 +1,88 @@
+import { useState } from 'react'
+import { validateForm, validateSingleField } from 'LoginPage/utils/formValidation'
+
+const PASSWORD = 'user[password]'
+const PASSWORD_CONFIRMATION = 'user[password_confirmation]'
+const validationConstraints = {
+  [PASSWORD]: {
+    presence: true,
+    length: { minimum: 1 }
+  },
+  [PASSWORD_CONFIRMATION]: {
+    presence: true,
+    length: { minimum: 1 },
+    equality: PASSWORD
+  }
+}
+
+const isFormValid = (formNode) => validateForm(formNode, validationConstraints)
+
+const getPasswordConfirmationError = (errors) => {
+  const passwordConfirmationError = (errors && errors[PASSWORD_CONFIRMATION]) && errors[PASSWORD_CONFIRMATION][0]
+  const passwordsMatchError = passwordConfirmationError && passwordConfirmationError.includes('is not equal to')
+  return {
+    passwordConfirmationError: !!passwordConfirmationError,
+    errorMessage: passwordsMatchError ? 'mustMatch' : 'isMandatory'
+  }
+}
+
+const comparePasswordsLength = (event) => {
+  const passwordInput = event.currentTarget['user_password']
+  const passwordConfirmationInput = event.currentTarget['user_password_confirmation']
+  return passwordConfirmationInput.value.length >= passwordInput.value.length
+}
+
+const useFormState = () => {
+  const [isFormDisabled, setIsFormDisabled] = useState(true)
+  const [passwordValue, setPasswordValue] = useState('')
+  const [isPasswordValid, setIsPasswordValid] = useState(undefined)
+  const [passwordConfirmationValue, setPasswordConfirmationValue] = useState('')
+  const [isPasswordConfirmationValid, setIsPasswordConfirmationValid] = useState(undefined)
+  const [confirmationErrorMessage, setConfirmationErrorMessage] = useState('isMandatory')
+  const [validationStarted, setValidationStarted] = useState(false)
+
+  const onFormChange = (event) => {
+    const errors = isFormValid(event.currentTarget)
+    const { passwordConfirmationError, errorMessage } = getPasswordConfirmationError(errors)
+    const initValidation = comparePasswordsLength(event)
+    if (initValidation && !validationStarted) {
+      setValidationStarted(initValidation)
+    }
+    if (initValidation || validationStarted) {
+      setIsPasswordConfirmationValid(!passwordConfirmationError)
+    }
+    setConfirmationErrorMessage(errorMessage)
+    setIsFormDisabled(!!errors)
+  }
```

Comment:
I still believe the `validationStarted` is not needed.
This would be the same:

```js
  const onFormChange = (event) => {
    const errors = isFormValid(event.currentTarget)
    const { passwordConfirmationError, errorMessage } = getPasswordConfirmationError(errors)
    const initValidation = comparePasswordsLength(event)
   
    if (initValidation) {
      setIsPasswordConfirmationValid(!passwordConfirmationError)
    }
    setConfirmationErrorMessage(errorMessage)
    setIsFormDisabled(!!errors)
  }
```

Reply:
It's not the same because one edge case is missing: 

1. You type password: "abc"
2. You type confirmation: "abcd" // Error is displayed
3. You delete some chars in confirmation: "ab" // Error not displayed anymore, as it's length is < password length

Error message should not disappear once validation has started

Reply:
@damianpm is it just me or it feels like the description of the edge case is clashing with your last sentence?
I would say: If a confirmation error is triggered, it will stay visible until correct pass confirmation is typed

Does that make sense to you?

Reply:
Yes, that's what I mean
Point 3 is what is we don't want
> You delete some chars in confirmation: "ab" // Error not displayed anymore, as it's length is < password length